### PR TITLE
resolve #5

### LINKED ISSUE #5: Cannot complete
Trying to solve this problem : 
Try to create an EventBus using callbag but cannot stop it when sending 2.

```javascript
const makeSubject = require('callbag-subject')
const subscribe = require('callbag-subscribe')
const { pipe } = require('callbag-basics')

const EventBus = makeSubject()

const observer = {
  next: data => console.log( `DATA: ${data}` ),
  error: err => console.error( `ERROR: ${err}` ),
  complete: () => console.log( 'Done!' )
}

pipe(
  EventBus,
  subscribe( observer )
)

setInterval(() => { EventBus(1, 'some data')}, 1000)
setTimeout(() => { EventBus(2) }, 5000)
```

Console display : 
```bash
DATA: some data
DATA: some data
DATA: some data
DATA: some data
DATA: some data
Done!
DATA: some data
DATA: some data
DATA: some data
DATA: some data
DATA: some data
DATA: some data
...
```

[Try online](https://stackblitz.com/edit/js-26ejca)

### PR DESCRIPTION
**Pull request by loreanvictor**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,8 +18,12 @@ function makeSubject() {
         sink = zinkz[i];
         if (sinks.indexOf(sink) > -1) sink(type, data);
       }
+      if (type === 2) {
+        done = true;
+        sinks.length = 0;
```
Comment (staltz) [inline index.js:23]: Could we not have a mutation of the array `length` property? Seems very unconventional to do such mutation of a property we don't own, and I also think we don't need it here to achieve #5.

**Review by staltz — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (loreanvictor) [reply index.js:23]: how do we not own it? we own the list and the `length` property is, [per spec, a read/write property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length). I suspect this is the fastest way to clear it out (though `sinks.splice(0, sinks.length))` should work equally well to the best of my knowledge).

for needing it: yeah its just being super safe on maintained references. I suspect (but I might be mistaken) that if for example someone holds a reference to the subject after completing it and we do not clear `sinks`, then references to all sinks who did not terminate on their own will also be maintained unnecessarily.

Comment (staltz) [reply index.js:23]: Okay today I learned something new. Didn't know that mutating `length` changes the actual size of the array.

**Comment by staltz** [comment]
Released 2.0.0

**Pull request merged**